carbonmark/lib/utils/string.utils.ts

```diff
@@ -0,0 +1,15 @@
+import { isStringArray } from "./types.utils";
+
+/**
+ * Joins an array of strings into a single string.
+ * If the input is already a string, it returns the input as is.
+ */
+export const joinArray = (value: string | string[]): string =>
+  isStringArray(value) ? value.join(",") : value;
+
+/**
+ * Checks if a string is empty and returns undefined if true.
+ * If the string is not empty, it returns the string as is.
+ */
+export const emptyToUndefined = (value: string): string | undefined =>
+  value === "" ? undefined : value;
```

Comment:
I think what we are trying to do here in the scope of this PR is filter empty strings right? Mapping them to undefined seems like a detour that also gets us there but I believe filter is what we should use.

Reply:
Sort of, this utility on it's own can be useful but in the context of the query params we're trying to remove attributes from the object that have the "" value. Perhaps we should be using `omitBy` instead?

Reply:
Ok I've updated to hopefully more clearly sanitise the query params:

1. Remove nullish values from arrays & nullish values from the object
2. Join any arrays using ","

This should cover more cases and be a little clearer